src/main/java/com/desiato/puresynth/services/UserService.java

```diff
@@ -16,13 +20,15 @@ public class UserService {
 
     private final UserRepository userRepository;
     private final SessionService sessionService;
+    private final ProjectConfig projectConfig;
 
     public List<User> getAllUsers() {
         return userRepository.findAll();
     }
 
-    public Optional<User> getUserById(Long id) {
-        return userRepository.findById(id);
+    public User getUserById(Long id) {
```

Comment:
I would recommend 2 methods:
- Optional<User> getUserById
- User getUserByIdOrThrow
Then when using your API, you can decide what to do and which method to pick.